extension/src/NABfunctions.ts

```diff
@@ -151,9 +154,28 @@ export async function findTranslatedTexts() {
                 throw new Error('This line does not contain any translated property or label.');
             }
             const textToSearchFor = selectedMlObject[0].xliffId();
-            let fileFilter = '';
-            if (Settings.getConfigSettings()[Setting.SearchOnlyXlfFiles] === true) { fileFilter = '*.xlf'; }
-            await VSCodeFunctions.findTextInFiles(textToSearchFor, false, fileFilter);
+            let langFiles = (await WorkspaceFunctions.getLangXlfFiles(vscode.window.activeTextEditor.document.uri));
+            let doSearch = true;
+            if (langFiles.length === 1) {
+                let langContent = fs.readFileSync(langFiles[0].fsPath, 'utf8');
```

Comment:
Line 160 to 167 could be a function.

Reply:
I would say that more than that! ;)
I'll refactor a bit...